manual/manual/refman/typedecl.etex

```diff
@@ -200,6 +204,18 @@ constraints, the variance properties of the type constructor
 are inferred from its definition, and the variance annotations are
 only checked for conformance with the definition.
 
+Injectivity annotations only make sense for abstract types and private
+row types, since they can otherwise be deduced from the type
+declaration: all parameters are injective for record and variant type
+declarations (including extensible types); for type abbreviations a
+parameter is injective if it has an injective occurrence in its
+defining equation (be it private or not). For constrained type
+parameters in type abbreviations, they are injective if either they
+appear at an injective position in the body, or if all their type
+variables are injective; in particular, if a constrained type
+parameter contains a variable that doesn't appear in the body, it
+cannot be injective.
```

Comment:
> if a constrained type parameter contains a variable that doesn't appear in the body, it cannot be injective.

I'm not sure about this criterion.  Isn't the following injective, even though `'b` doesn't appear in the body?

```ocaml
type !'a t = 'a constraint 'a = 'b
```

Or does "doesn't appear in the body" mean something else?

Reply:
The constraints makes `'b` equal to `'a`, so it clearly appears in the body.